spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -205,15 +219,20 @@ public StructType readSchema() {
     // broadcast the table metadata as input partitions will be sent to executors
     Broadcast<Table> tableBroadcast = sparkContext.broadcast(SerializableTable.copyOf(table));
 
-    List<InputPartition<ColumnarBatch>> readTasks = Lists.newArrayList();
-    for (CombinedScanTask task : tasks()) {
-      readTasks.add(new ReadTask<>(
-          task, tableBroadcast, expectedSchemaString, caseSensitive,
-          localityPreferred, new BatchReaderFactory(batchSize)));
-    }
-    LOG.info("Batching input partitions with {} tasks.", readTasks.size());
+    List<CombinedScanTask> scanTasks = tasks();
+    InputPartition<ColumnarBatch>[] readTasks = new InputPartition[scanTasks.size()];
 
-    return readTasks;
+    Tasks.range(readTasks.length)
+        .stopOnFailure()
+        .executeWith(readTasksInitExecutorService)
+        .run(index -> {
```

Comment:
The block is unnecessary because there is only one expression, can you remove it?

Reply:
Done